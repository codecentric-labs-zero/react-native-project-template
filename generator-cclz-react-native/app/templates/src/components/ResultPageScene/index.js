import React from 'react'
import { View, Text } from 'react-native'

export default ResultPageScene = (props) => (
  <View style={{flex: 1, flexDirection: 'column', alignItems: 'center', justifyContent: 'center'}}>
    <Text onPress={props.onBackButtonClick}>Back</Text>
    <Text onPress={props.onImmediateActionClick}>Trigger immediate action</Text>
    <Text onPress={props.onAsyncActionClick}>Trigger async action</Text>
    <Text>Result: {props.result}</Text>
  </View>
)
