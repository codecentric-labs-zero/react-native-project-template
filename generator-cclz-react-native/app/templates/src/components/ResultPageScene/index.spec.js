import 'react-native';
import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ResultPageScene from './index';

const textIs = (text) => (
  (e) => e.props().children === text
);

describe('ResultPageScene', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <ResultPageScene />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onBackButtonClick when clicking back', () => {
    const onButtonClick = jest.fn();
    const element = shallow(<ResultPageScene onBackButtonClick={onButtonClick} />);
    element.findWhere(textIs('Back')).simulate('press');
    expect(onButtonClick).toBeCalled();
  });

  it('calls onImmediateActionClick when clicking trigger immediate action', () => {
    const onImmediateActionClick = jest.fn();
    const element = shallow(<ResultPageScene onImmediateActionClick={onImmediateActionClick} />);
    element.findWhere(textIs('Trigger immediate action')).simulate('press');
    expect(onImmediateActionClick).toBeCalled();
  });

  it('calls onAsyncActionClick when clicking trigger async action', () => {
    const onAsyncActionClick = jest.fn();
    const element = shallow(<ResultPageScene onAsyncActionClick={onAsyncActionClick} />);
    element.findWhere(textIs('Trigger async action')).simulate('press');
    expect(onAsyncActionClick).toBeCalled();
  });

  it('shows text for result', () => {
    const result = 'the result';
    const element = shallow(<ResultPageScene result={result} />);
    expect(element.containsMatchingElement(<Text>Result: {result}</Text>)).toBeTruthy();
  })
})
