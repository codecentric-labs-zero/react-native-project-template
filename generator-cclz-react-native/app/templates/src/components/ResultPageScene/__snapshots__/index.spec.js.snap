exports[`ResultPageScene renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={undefined}>
    Back
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={undefined}>
    Trigger immediate action
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={undefined}>
    Trigger async action
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Result: 
  </Text>
</View>
`;
